server/controllers/finance/reports/break_even_fee_center/index.js

```diff
@@ -0,0 +1,147 @@
+const q = require('q');
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+
+const TEMPLATE = './server/controllers/finance/reports/break_even_fee_center/report.handlebars';
+const AccountReference = require('../../accounts').references;
+const setting = require('./setting');
+const getDistributionKey = require('../../distributionFeeCenter/getDistributionKey');
+
+// expose to the API
+exports.report = report;
+
+// default report parameters
+const DEFAULT_PARAMS = {
+  csvKey : 'brea_report',
+  filename : 'TREE.BREAK_EVEN_REPORT',
+  orientation : 'portrait',
+  footerRight : '[page] / [toPage]',
+};
+
+/**
+ * @function report
+ *
+ * @description
+ * This function renders the balance of accounts references as report.  The account_reference report provides a view
+ * of the balance of account_references for a given period of fiscal year.
+ */
+function report(req, res, next) {
+  const params = req.query;
+  const feeCentersChecked = [];
+
+  if (params.defineFeeCenters && params.feeCenters) {
+    params.feeCenters.forEach(item => {
+      if (parseInt(item, 10)) {
+        feeCentersChecked.push(item);
+      }
+    });
+  }
+
+  const data = {};
+  let reporting;
+
+  params.start_date = new Date(params.start_date);
+  params.end_date = new Date(params.end_date);
+
+  data.period = {
+    start_date : params.start_date,
+    end_date : params.end_date,
+    fiscalYearStart : params.fiscalYearStart,
+  };
+
+  _.defaults(params, DEFAULT_PARAMS);
+
+  try {
+    reporting = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    next(e);
+    return;
+  }
+
+  const getFeeCenter = `
+    SELECT fc.id, fc.label, fc.is_principal FROM fee_center AS fc ORDER BY fc.label ASC, fc.is_principal DESC
+  `;
+
+  const getFeeCenterReference = `
+    SELECT fc.id, fc.label, fc.is_principal, rf.fee_center_id, rf.account_reference_id, 
+    rf.is_cost, rf.is_variable, rf.is_turnover, ar.abbr
+    FROM fee_center AS fc
+    JOIN reference_fee_center AS rf ON rf.fee_center_id = fc.id
+    JOIN account_reference AS ar ON ar.id = rf.account_reference_id
+    ORDER BY fc.label
+  `;
+
+  const getFeeCenterDistribution = `
+    SELECT fcd.is_cost, fcd.is_variable, fcd.is_turnover, BUID(fcd.row_uuid) AS row_uuid,
+    fca.label AS auxiliary, fcp.label AS principal, SUM(fcd.debit_equiv) AS debit,
+    SUM(fcd.credit_equiv) AS credit, gl.trans_date, fcd.auxiliary_fee_center_id,
+    fcd.principal_fee_center_id
```

Comment:
Because you are grouping by `fcd.auxiliary_fee_center_id` then `fcd.principal_fee_center_id`, let's put those two first in the SELECT statement.